src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsSourceView.cs

```diff
@@ -0,0 +1,296 @@
+				var bindableObservableVector = m_vector as IBindableObservableVector;
+				if (bindableObservableVector != null)
+				{
+					bindableObservableVector.VectorChanged += OnBindableVectorChanged;
+					m_bindableObservableVector = bindableObservableVector;
+				}
+				else
+				{
+					var observableVector = m_vector as IObservableVector<object>;
+					if (observableVector != null)
+					{
+						observableVector.VectorChanged += OnVectorChanged;
+						m_observableVector = observableVector;
+					}
+				}
+			}
+		}
+
+		private void OnCollectionChanged(
+					object sender,
+					NotifyCollectionChangedEventArgs e)
+		{
+			OnItemsSourceChanged(e);
+		}
+
+		void OnBindableVectorChanged(IBindableObservableVector sender, object e)
+		{
+			OnVectorChanged(
+				sender as IObservableVector<object>,
+				e as IVectorChangedEventArgs);
+		}
+
+		void OnVectorChanged(
+			IObservableVector<object> sender,
+			IVectorChangedEventArgs e)
+		{
+			// We need to build up NotifyCollectionChangedEventArgs here to raise the event.
+			// There is opportunity to make this faster by caching the args if it does 
+			// show up as a perf issue.
+			// Also note that we do not access the data - we just add nullptr. We just 
+			// need the count.
+
+			NotifyCollectionChangedAction action;
+			int oldStartingIndex = -1;
+			int newStartingIndex = -1;
+
+			var oldItems = new List<object>();
+			var newItems = new List<object>();
+
+			switch (e.CollectionChange)
+			{
+				case CollectionChange.ItemInserted:
+					action = NotifyCollectionChangedAction.Add;
+					newStartingIndex = (int)e.Index;
+					newItems.Add(null);
+					OnItemsSourceChanged(new NotifyCollectionChangedEventArgs(action, newItems, newStartingIndex));
+					break;
+				case CollectionChange.ItemRemoved:
+					action = NotifyCollectionChangedAction.Remove;
+					oldStartingIndex = (int)e.Index;
+					oldItems.Add(null);
+					OnItemsSourceChanged(new NotifyCollectionChangedEventArgs(action, oldItems, oldStartingIndex));
+					break;
+				case CollectionChange.ItemChanged:
+					action = NotifyCollectionChangedAction.Replace;
+					oldStartingIndex = (int)e.Index;
+					newStartingIndex = oldStartingIndex;
+					newItems.Add(null);
+					oldItems.Add(null);
+					OnItemsSourceChanged(new NotifyCollectionChangedEventArgs(action, newItems, oldItems, newStartingIndex));
+					break;
+				case CollectionChange.Reset:
+					action = NotifyCollectionChangedAction.Reset;
+					OnItemsSourceChanged(new NotifyCollectionChangedEventArgs(action));
+					break;
+				default:
+					throw new InvalidOperationException("Unsupported collection change");
+			}
+
+			// Uno specific: WinUI uses an internal overload of NotifyCollectionChangedEventArgs constructor with 5 args,
+			// we use a specific overload for each case instead
+
+			//OnItemsSourceChanged(
+			//	new NotifyCollectionChangedEventArgs(
+			//		action,
+			//		newItems,
+			//		oldItems,
+			//		newStartingIndex,
+			//		oldStartingIndex);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)